internal/metastore/kv/kv_catalog.go

```diff
@@ -28,60 +28,185 @@ import (
+		keys = append(keys, k)
+		values = append(values, v)
+	}
+	min := func(a, b int) int {
+		if a < b {
+			return a
+		}
+		return b
+	}
+	for i := 0; i < len(kvs); i = i + maxTxnNum {
+		end := min(i+maxTxnNum, len(keys))
+		batch, err := buildKvs(keys[i:end], values[i:end])
+		if err != nil {
+			return err
+		}
+		// TODO: atomicity is not promised. Garbage will be generated.
+		if err := snapshot.MultiSave(batch, ts); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
 func (kc *Catalog) CreateCollection(ctx context.Context, coll *model.Collection, ts typeutil.Timestamp) error {
-	k1 := fmt.Sprintf("%s/%d", CollectionMetaPrefix, coll.CollectionID)
+	k1 := buildCollectionKey(coll.CollectionID)
 	collInfo := model.MarshalCollectionModel(coll)
 	v1, err := proto.Marshal(collInfo)
 	if err != nil {
 		log.Error("create collection marshal fail", zap.String("key", k1), zap.Error(err))
 		return err
 	}
 
-	// save ddOpStr into etcd
 	kvs := map[string]string{k1: string(v1)}
-	for k, v := range coll.Extra {
-		kvs[k] = v
+
+	// save partition info to newly path.
+	for _, partition := range coll.Partitions {
+		k := buildPartitionKey(coll.CollectionID, partition.PartitionID)
+		partitionInfo := model.MarshalPartitionModel(partition)
+		v, err := proto.Marshal(partitionInfo)
+		if err != nil {
+			return err
+		}
+		kvs[k] = string(v)
 	}
 
-	err = kc.Snapshot.MultiSave(kvs, ts)
-	if err != nil {
-		log.Error("create collection persist meta fail", zap.String("key", k1), zap.Error(err))
-		return err
+	// no default aliases will be created.
+
+	// save fields info to newly path.
+	for _, field := range coll.Fields {
+		k := buildFieldKey(coll.CollectionID, field.FieldID)
+		fieldInfo := model.MarshalFieldModel(field)
+		v, err := proto.Marshal(fieldInfo)
+		if err != nil {
+			return err
+		}
+		kvs[k] = string(v)
 	}
 
-	return nil
+	// TODO: atomicity should be promised outside.
+	maxTxnNum := 64
+	return batchSave(kc.Snapshot, maxTxnNum, kvs, ts)
 }
 
-func (kc *Catalog) CreatePartition(ctx context.Context, coll *model.Collection, ts typeutil.Timestamp) error {
-	k1 := fmt.Sprintf("%s/%d", CollectionMetaPrefix, coll.CollectionID)
-	collInfo := model.MarshalCollectionModel(coll)
-	v1, err := proto.Marshal(collInfo)
+func (kc *Catalog) loadCollection(ctx context.Context, collectionID typeutil.UniqueID, ts typeutil.Timestamp) (*pb.CollectionInfo, error) {
+	collKey := buildCollectionKey(collectionID)
+	collVal, err := kc.Snapshot.Load(collKey, ts)
 	if err != nil {
-		log.Error("create partition marshal fail", zap.String("key", k1), zap.Error(err))
-		return err
+		log.Error("get collection meta fail", zap.String("key", collKey), zap.Error(err))
+		return nil, err
 	}
 
-	kvs := map[string]string{k1: string(v1)}
-	err = kc.Snapshot.MultiSave(kvs, ts)
+	collMeta := &pb.CollectionInfo{}
+	err = proto.Unmarshal([]byte(collVal), collMeta)
+	return collMeta, err
+}
+
+func partitionVersionAfter210(collMeta *pb.CollectionInfo) bool {
```

Comment:
functions return bool better to named with "is" prefix.
you can change this in next pr.